### PR TITLE
Init the communication with kafka

### PR DESCRIPTION
**Pull request by sswguo**

### CODE REVIEW

**Comment by ruhan1** [comment]
This looks cool! 

Let's talk about the events type and flow right after we back from holidays. 

Mainly, the storage service is simple, only consumes events (storage/deletion for now). The other (like Metadata service) send the file operation request. 

I am thinking a typical eg, 
1. Indy or promote service upload one or a list of pom file(s), via direct pathmap lib, when finish. it send FileStoreEvent(s), 
2. Metadata service get it and calcaulate what metadata files to clean, it sends a FileDeleteRequest message, 
3. Storage service consumes it and do the deletion(s)

The opposite flow is when Indy asks for a metadata and but not found, it then ask Metadata service. Metadata service ask Storage service for a list of pom files and generate the metadata file. Once done, it will send the file to Indy, meanwhile write file to Storage service via REST api.

**Comment by sswguo** [comment]
Yeah, it's important to figure the event flow out. And I believe John had listed most of that: https://docs.engineering.redhat.com/pages/viewpage.action?spaceKey=MPAKT&title=Indy+Events

Another thing is how to do ser/der of the event object, I found one blog to explain the use of Avro, that provides schema-based, robust, and fast binary serialization. But it needs to bring a scheme registry, so not sure if it worth to do it. 
https://quarkus.io/blog/kafka-avro/
( In the PR I used the objectMapper way to do the serialization and it works in my local test. )

**Comment by ruhan1** [comment]
What I wonder is who to send the FileEvent. On the chart, it is File service. But maybe the Content retrieval service is the right one. As it use the pathmap lib directly.
Yep, we will figure out the ser/des next.

File: src/main/java/org/commonjava/service/storage/event/FileEvent.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package org.commonjava.service.storage.event;
+
+public class FileEvent
+{
+
+    private String tackingID;
```
Comment (ligangty) [inline src/main/java/org/commonjava/service/storage/event/FileEvent.java]: tackingID->trackingID?

**Review by ligangty — COMMENTED** [review]
Looks cool. This is a pure start for the event support, maybe we can also see how to setup the UT & IT for this kafka event support. (Not sure if there is example in quarkus tutorials)

File: src/main/java/org/commonjava/service/storage/event/FileEvent.java (hunk 1 above)
Comment (sswguo) [reply src/main/java/org/commonjava/service/storage/event/FileEvent.java]: Yeah. 👍

**Comment by sswguo** [comment]
> Looks cool. This is a pure start for the event support, maybe we can also see how to setup the UT & IT for this kafka event support. (Not sure if there is example in quarkus tutorials)

There is a repo that we can ref for how to setup the tests: https://github.com/quarkusio/quarkus/tree/master/integration-tests

**Pull request merged**